### DISCUSSION TITLE
Simultanious sin(x) and cos(x) calculation.

### DISCUSSION THREAD

**Discussion by QuantifEye** [dotnet/runtime#47694] [Ideas] [answered]
I am not sure about arm, but x86 has an instruction to calculate sine and cosine in one step.

see `https://mudongliang.github.io/x86/html/file_module_x86_id_115.html`

It would be great if function `public static double SinCos(double x, out double cos);` could be added to the `System.Math` class that calculates sine, and return cosine as an out parameter.

**Comment by PathogenDavid** [comment, marked as answer]
`Math.SinCos` and `MathF.SinCos` were added in https://github.com/dotnet/runtime/pull/46551 which I would assume will ship with .NET 6.

**Reply by tannergooding** [reply]
Correct. Likewise, `fsincos` itself is not a very efficient instruction and would itself likely be slower than just calling `Math.Sin` and `Math.Cos` independently.
